src/context.c

```diff
@@ -36,8 +45,62 @@ fix_ipv6_string(char *ipv6_str)
     }
 }
 
+#ifdef USE_XTT
 int
-enftun_context_init(struct enftun_context* ctx)
+enftun_context_init_xtt(struct enftun_context* ctx)
```

Comment:
I agree the duplication now seems unnecessary. I would probably go the route of taking the xtt struct out of the `enftun_context` since everything in the `struct enftun_xtt` is strictly related to the `enftun_xtt_handshake`.

Reply:
Sounds good.  Let's take that route then.